ui/components/app/confirm-page-container/confirm-page-container.component.js

```diff
@@ -235,9 +247,70 @@ export default class ConfirmPageContainer extends Component {
               hideTitle={hideTitle}
               supportsEIP1559V2={supportsEIP1559V2}
               hasTopBorder={showAddToAddressDialog}
+              currentTransaction={currentTransaction}
+              nativeCurrency={nativeCurrency}
+              networkName={networkName}
+              showBuyModal={showBuyModal}
             />
           )}
-          {shouldDisplayWarning && (
+          {shouldDisplayWarning && errorKey === INSUFFICIENT_FUNDS_ERROR_KEY && (
+            <div className="confirm-approve-content__warning">
+              {currentTransaction.chainId === '0x1' ? (
+                <ActionableMessage
+                  message={
+                    <Typography
+                      variant={TYPOGRAPHY.H7}
+                      align="left"
+                      margin={[0, 0]}
+                    >
+                      {this.context.t('insufficientCurrency', [
+                        nativeCurrency,
+                        networkName,
+                      ])}
+                      <Button
+                        key="link"
+                        type="secondary"
+                        className="confirm-approve-content__warning__link"
+                        onClick={() => showBuyModal()}
+                        style={{
+                          color: '#037dd6',
+                          padding: 0,
+                          fontSize: '12px',
+                        }}
```

Comment:
I think we generally want to avoid inline styles like this.

Reply:
removed it.